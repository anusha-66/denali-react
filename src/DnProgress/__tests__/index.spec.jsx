/**
 * Copyright 2020, Verizon Media
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */

import React from 'react';
import { mount, render } from 'enzyme';
import { DnProgress, DnProgressSizes, DnProgressBgStatus } from '../../index'; // src/index.tsx

describe('DnProgress Tests', () => {
    it('should export DnProgressSizes', () => {
        expect.assertions(1);

        expect(DnProgressSizes).toStrictEqual({
            default: '',
            small: 'sm',
            medium: 'md',
            large: 'lg',
        });
    });

    it('should export DnProgressBgStatus', () => {
        expect.assertions(1);

        expect(DnProgressBgStatus).toStrictEqual({
            default: '',
            danger: 'danger',
            warning: 'warning',
            success: 'success',
        });
    });

    it('should render with default props', () => {
        expect.assertions(1);

        const rendered = render(<DnProgress dataList={[{ key: 'key1', value: 50 }]} />);

        expect(rendered).toMatchSnapshot();
    });

    it('should render with isLoading props', () => {
        expect.assertions(1);

        const rendered = render(<DnProgress dataList={[{ key: 'key1', value: 0 }]} isLoading />);

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with custom props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnProgress
                dataList={[{ key: 'key1', value: 25, bgStatus: DnProgressBgStatus.danger, className: 'foo-1' }]}
                size={DnProgressSizes.large}
                className="p-18"
            />,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render multi bar correctly with custom props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnProgress
                dataList={[
                    { key: 'key1', value: 15, bgStatus: DnProgressBgStatus.danger, className: 'foo-1' },
                    { key: 'key2', value: 30, bgStatus: DnProgressBgStatus.warning, className: 'foo-2' },
                    { key: 'key3', value: 20, bgStatus: DnProgressBgStatus.success, className: 'foo-3' },
                ]}
                size={DnProgressSizes.large}
                className="p-18"
            />,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should call on click correctly', () => {
        expect.assertions(1);

        const mockedOnClick = jest.fn();
        const wrapper = mount(<DnProgress dataList={[{ key: 'key1', value: 25 }]} onClick={mockedOnClick} />);
        wrapper.find('div.d-progress').simulate('click');

        expect(mockedOnClick).toHaveBeenCalledTimes(1);
    });
});
