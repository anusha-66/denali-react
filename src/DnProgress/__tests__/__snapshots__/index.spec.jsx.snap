// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DnProgress Tests should render correctly with custom props 1`] = `
<div
  class="d-progress d-progress__lg p-18"
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="25"
    class="d-progress--bar has-bg-status-danger foo-1"
    role="progressbar"
    style="width:25%"
  />
</div>
`;

exports[`DnProgress Tests should render multi bar correctly with custom props 1`] = `
<div
  class="d-progress d-progress__lg p-18"
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="15"
    class="d-progress--bar has-bg-status-danger foo-1"
    role="progressbar"
    style="width:15%"
  />
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="30"
    class="d-progress--bar has-bg-status-warning foo-2"
    role="progressbar"
    style="width:30%"
  />
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="20"
    class="d-progress--bar has-bg-status-success foo-3"
    role="progressbar"
    style="width:20%"
  />
</div>
`;

exports[`DnProgress Tests should render with default props 1`] = `
<div
  class="d-progress"
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="50"
    class="d-progress--bar"
    role="progressbar"
    style="width:50%"
  />
</div>
`;

exports[`DnProgress Tests should render with isLoading props 1`] = `
<div
  class="d-progress d-progress__loading"
/>
`;
